juriscraper/lib/network_utils.py

```diff
@@ -283,3 +285,73 @@ def get_attachments(
             self._map_docket_entry_document(attachment)
             for attachment in raw_attachments
         ]
+
+    def download_pdf(
+        self,
+        document_id: str,
+        page_count: int,
+        document_url: str,
+        max_attempts: int = 60,
+        poll_interval: float = 2.0,
```

Comment:
I think a default value of 2 seconds between requests might be too large. Did you notice that this usually takes more than 2 seconds? Perhaps 1.0 would be better?

And increase the `max_attempts`?

Reply:
> Did you notice that this usually takes more than 2 seconds? 

 I've observed that the full process, including the merge request and download, typically takes two seconds, but one second is a better, more conservative interval.